lib/beautydate_api/api_resource.rb

```diff
@@ -34,15 +27,12 @@ def relative_url(options=nil)
       end
 
       def object_base_uri
-        pluralized_models = %w'business apiconsumer'
-        pluralized_models_names = %w'businesses consumers'
-
-        plural_position = pluralized_models.index(self.object_type)
-        if plural_position
-          pluralized_models_names[plural_position]
-        else
-          self.object_type + 's'
-        end
+        self.name
+          .to_s
+          .demodulize
+          .tableize
+          .titleize
+          .parameterize
```

Comment:
Por que é necessário usar o `titleize` antes do `parameterize`?
O parameterize já não tira todas as letras maiúsculas?

Acho que era uma boa colocar comentários após cada uma das linhas mostrando um caso típico pra vermos todas as transformações que estão acontecendo na `string`.

Ah, essa manipulação toda de strings deve ser meio lenta, então se for chamar esse método várias vezes - por exemplo, pra manipular uma coleção - talvez seja interessante fazer um cache numa variável de classe, já que esse valor é constante.